### ISSUE TITLE
[Bug] copied style from HTML are not kept

### ISSUE THREAD

**Issue by hj-ueda** [Milkdown/milkdown#956]
### Initial checklist

- [X] I agree to follow the [code of conduct](https://github.com/Saul-Mirone/milkdown/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched [issues](hhttps://github.com/Saul-Mirone/milkdown/issues) and [discussions](https://github.com/Saul-Mirone/milkdown/discussions) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

6.5.4 - 7.1.1

### Link to runnable example

https://stackblitz.com/edit/github-txeh7z?file=index.html

### Steps to reproduce

try in above sample
![demo](https://user-images.githubusercontent.com/106786793/228521274-be1173b5-be3a-4f38-b1d3-1002cfe74514.gif)

### Expected behavior

The syles of character's with some symbols on HTML are should be kept when they are copied and pasted to Milkdown Editor.

### Actual behavior

Some patterns of style are escaped in Milkdown Editor as I showed in the example .
(This bug couldn't be reproduced in v6.1.4. I tried in https://codesandbox.io/s/milkdown-i18n-2uzcvk)

### Runtime

Chrome

### OS

Windows

### Build and bundle tools

Vite

**Comment by Saul-Mirone** [comment]
This is an interesting issue. I'll talk about the behavior in milkdown v7.

Let's take a look at your first example.

What would we expect this markdown should be rendered as?

```markdown
**abc-**def
```

Unfortunately, you will get:

> **abc-**def

This is same in both commonmark and gfm. Markdown won't consider it as a bold style because of it's language specification.

So, what's the problem if Milkdown still render it as bold?

You'll find that when you load this content from Markdown string, it sill display as

> **abc-**def

instead of

> **abc**-def

That's definitely wrong, right? After all you just save the content and reload it, everything changed. You can put `**abc-**def` as `defaultValue` in any version of milkdown. You'll know what I'm saying.

---

In short: this is not a bug, it's a feature to avoid invalid markdown syntax. It can help you find your syntax error during edit.
